woocommerce/woocommerce-ios
# [Blaze] Update campaign status options

Closes: #12120 


## Description

v1.1. GET campaigns endpoint sends different values for `status` field. 

From the API documentation, the following are the values for the `status` field.

> “pending” “scheduled” “active” “rejected” “canceled” “finished”

Internal - peeHDf-2fw-p2#comment-1896

This PR removes old status values and adds the new values to match the backend. 


Changes
- Update the values and removes unused cases from `Status` enum.
- Assign title and colors for new status values.
- Use new status values to determine whether a product has existing campaigns. 

## Testing instructions
- Login into a Woo store eligible for Blaze.
- Navigate to the Products tab and tap on a published product with no Blaze campaigns.
- You should see the "Promote with Blaze" button.
- Navigate back and open a product with an existing campaign. (Campaign can be newly submitted or currently running)
- You should not see the "Promote with Blaze" button in the product detail screen.

## Screenshots

| With pending campaign | No campaign |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-03-05 at 10 16 40](https://github.com/woocommerce/woocommerce-ios/assets/524475/f5f1b89c-6a48-4e56-bcc0-7838c40cce3a) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-03-05 at 10 16 44](https://github.com/woocommerce/woocommerce-ios/assets/524475/f8aff7bf-06f4-42d4-aac1-9553a63e498b) | 

---
